### PR TITLE
Initial Streamlit callback integration doc (md)

### PR DESCRIPTION
**Pull request by sfc-gh-jcarroll**
**Description:** Add a documentation page for the Streamlit Callback Handler integration (#6315)

Notes:
- Implemented as a markdown file instead of a notebook since example code runs in a Streamlit app (happy to discuss / consider alternatives now or later)
- Contains an embedded Streamlit app -> https://mrkl-minimal.streamlit.app/ Currently this app is hosted out of a Streamlit repo but we're working to migrate the code to a LangChain owned repo

streamlit_docs

cc @dev2049 @tconkling

### CODE REVIEW

File: docs/extras/modules/callbacks/integrations/streamlit.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+[API reference](https://api.python.langchain.com/en/latest/modules/callbacks.html#langchain.callbacks.StreamlitCallbackHandler).
+
+### Scenario 1: Using an Agent with Tools
+
+The primary supported use case today is visualizing the actions of an Agent with Tools (or Agent Executor). You can create an
+agent in your Streamlit app and simply pass the `StreamlitCallbackHandler` to `agent.run()` in order to visualize the
+thoughts and actions live in your app.
+
+```python
+from langchain import OpenAI
```
Comment (dev2049) [inline docs/extras/modules/callbacks/integrations/streamlit.md:47]: über nit: we're trying to move away from importing from package root, so would do `langchain.llms import OpenAI`

File: docs/extras/modules/callbacks/integrations/streamlit.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+from langchain.agents import AgentType, initialize_agent, load_tools
+from langchain.callbacks import StreamlitCallbackHandler
+import streamlit as st
+
+llm = OpenAI(temperature=0, streaming=True)
+tools = load_tools(["serpapi"], llm=llm)
```
Comment (dev2049) [inline docs/extras/modules/callbacks/integrations/streamlit.md:53]: up to you, but have started using duck duck go (name is "duckduckgo_search") instead, just bc it doesn't require signup/api key

**Review by dev2049 — APPROVED** [review]
two optional comments, but generally lgtm! just lemme know when you want me to land

File: docs/extras/modules/callbacks/integrations/streamlit.md (hunk 2 above)
Comment (sfc-gh-jcarroll) [reply docs/extras/modules/callbacks/integrations/streamlit.md]: Nice! Will have to use that in other examples too :)

**Pull request merged**